### PR TITLE
Add sandbox config

### PR DESCRIPTION
**Pull request by rut31337**
##### SUMMARY

Add new sandbox config.  This is a sandbox configuration that does not install anything.  It just gives the user a cloud account.

##### ISSUE TYPE
- New config Pull Request

##### COMPONENT NAME
config sandbox

### CODE REVIEW

File: ansible/configs/sandbox/default_vars.yml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+install_student_user: false
+install_ftl: false
+
+# install openshift python modules on bastion
+install_k8s_modules: true
+
+## Automatically deploy ocp4 cluster using the installer
+install_ocp4: false
+ocp4_installer_version: "4.1.0"
+cluster_name: "cluster-{{ guid }}"
```
Comment (jkupferer) [inline ansible/configs/sandbox/default_vars.yml:37]: Thes variables are not relevant, right?

File: ansible/configs/sandbox/default_vars.yml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+# Smoke tests
+# if set to true, the playbook will fail if the smoke tests failed.
+# if set to false, the smoke tests are just informative.
+smoke_tests: false
+
+## Install Authentication.
+## Valid options are none, htpasswd, ldap (and maybe in the future sso)
+## For LDAP a bindPassword needs to be passed via the command line
+install_idm: "none"
+idm_ca_url: https://gpte-public.s3.amazonaws.com/opentlc_ipa_ca.crt
```
Comment (jkupferer) [inline ansible/configs/sandbox/default_vars.yml:48]: These too? This is for OpenShift config I believe

File: ansible/configs/sandbox/destroy_cluster.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+- debug:
+    msg: "Deleting {{ cluster_dir }} using user {{ username }}"
+
+- name: destroy terraform resources (openshift-install destroy cluster)
+  environment:
+    PATH: /usr/local/sbin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin:/usr/local/bin:/usr/local/bin:/root/bin
+  shell: openshift-install destroy cluster --dir={{ cluster_dir }}
```
Comment (jkupferer) [inline ansible/configs/sandbox/destroy_cluster.yml:8]: `openshift-install`?

**Review by jkupferer — CHANGES_REQUESTED** [review]
It looks like there is a lot in this PR that is copied over from an openshift config?

File: ansible/configs/sandbox/default_vars.yml (hunk 1 above)
Comment (rut31337) [reply ansible/configs/sandbox/default_vars.yml:19]: updated

File: ansible/configs/sandbox/default_vars.yml (hunk 2 above)
Comment (rut31337) [reply ansible/configs/sandbox/default_vars.yml]: updated

File: ansible/configs/sandbox/destroy_cluster.yml (hunk 3 above)
Comment (rut31337) [reply ansible/configs/sandbox/destroy_cluster.yml]: cleaned up

File: ansible/configs/sandbox/files/cloud_providers/ec2_cloud_template.j2 [hunk 4]
```diff
@@ -0,0 +1,402 @@
+#jinja2: lstrip_blocks: "True"
```
Comment (jkupferer) [inline ansible/configs/sandbox/files/cloud_providers/ec2_cloud_template.j2:1]: This seems odd? The sandbox requires a custom ec2_cloud_template.j2?

**Review by jkupferer — CHANGES_REQUESTED** [review]

File: ansible/configs/sandbox/default_vars.yml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+zone_internal_dns: "{{ guid }}.internal."
+chomped_zone_internal_dns: "{{ guid }}.internal"
+
+bastion_public_dns: "bastion.{{ subdomain_base }}."
+bastion_public_dns_chomped: "bastion.{{ subdomain_base }}"
+vpcid_name_tag: "{{ subdomain_base }}"
+
+az_1_name: "{{ aws_region }}a"
+az_2_name: "{{ aws_region }}b"
```
Comment (jkupferer) [inline ansible/configs/sandbox/default_vars.yml:91]: These `az_*_name` variables do not do anything in this config.

**Review by jkupferer — CHANGES_REQUESTED** [review]

File: ansible/configs/sandbox/default_vars.yml (hunk 5 above)
Comment (jkupferer) [reply ansible/configs/sandbox/default_vars.yml:91]: Also the `vpcid_name_tag` does not do anything.

File: ansible/configs/sandbox/default_vars.yml (hunk 1 above)
Comment (jkupferer) [reply ansible/configs/sandbox/default_vars.yml:19]: and `install_k8s_modules` doesn't do anything

File: ansible/configs/sandbox/default_vars.yml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+# install openshift python modules on bastion
+install_k8s_modules: true
+
+cluster_name: "cluster-{{ guid }}"
+
+# Smoke tests
+# if set to true, the playbook will fail if the smoke tests failed.
+# if set to false, the smoke tests are just informative.
+smoke_tests: false
```
Comment (jkupferer) [inline ansible/configs/sandbox/default_vars.yml:24]: Another var that does nothing

File: ansible/configs/sandbox/default_vars.yml [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+# you can use the key used to create the environment or use your own self generated key
+# if you set "use_own_key" to false your PRIVATE key will be copied to the bastion. (This is {{key_name}})
+
+use_own_key: true
+env_authorized_key: "{{ guid }}key"
+ansible_ssh_private_key_file: ~/.ssh/{{ key_name }}.pem
+set_env_authorized_key: true
+
+# Is this running from Red Hat Ansible Tower
+#tower_run: false
```
Comment (jkupferer) [inline ansible/configs/sandbox/default_vars.yml:37]: Should strip this out

File: ansible/configs/sandbox/destroy_cluster.yml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (jkupferer) [inline ansible/configs/sandbox/destroy_cluster.yml:1]: This whole file should go

File: ansible/configs/sandbox/destroy_env.yml [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+                "Name=instance-state-name,Values=pending,running,shutting-down,stopping,stopped"
+                --query 'Reservations[*].Instances[*].InstanceId'
+                --output text
+              changed_when: false
+              # Sometimes deployment fails before any OCP instances is created, so comment this line:
+              # failed_when: instanceids.stdout | trim | length == 0
+              register: instanceids
+              until: instanceids is succeeded
+              retries: 10
+              delay: "{{ 60|random(start=3, step=1) }}"
```
Comment (jkupferer) [inline ansible/configs/sandbox/destroy_env.yml:116]: This task is OpenShift specific and references `cluster_name`. Should drop it

File: ansible/configs/sandbox/destroy_env.yml [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+
+            - name: Start cluster instances
+              when: instanceids.stdout | trim | length > 0
+              command: >-
+                    aws ec2 start-instances --instance-ids {{ instanceids.stdout | trim }}
+              register: _startinstances
+              until: _startinstances is succeeded
+              retries: 10
+              delay: "{{ 60|random(start=3, step=1) }}"
```
Comment (jkupferer) [inline ansible/configs/sandbox/destroy_env.yml:125]: Likewise this task...

File: ansible/configs/sandbox/destroy_env.yml [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+            - name: Wait for clientVM instance
+              command: "aws ec2 wait instance-running --instance-ids '{{clientvm_id}}'"
+
+            - name: Wait for cluster instances
+              when: instanceids.stdout | trim | length > 0
+              ignore_errors: true
+              command: >-
+                    aws ec2 wait instance-running
+                    --filters "Name=tag:clusterid,Values={{ cluster_name }}"
```
Comment (jkupferer) [inline ansible/configs/sandbox/destroy_env.yml:135]: and here too

File: ansible/configs/sandbox/files/cloud_providers/ec2_cloud_template.j2 (hunk 4 above)
Comment (rut31337) [reply ansible/configs/sandbox/files/cloud_providers/ec2_cloud_template.j2]: updated

**Comment by rut31337** [comment]
ok i think im up to date with the requests

**Review by jkupferer — APPROVED** [review]

**Pull request merged**